tutorials/async_buffer_deadline/README.md

```diff
@@ -0,0 +1,133 @@
+
+1.  **Fixed `tx1` Period, Varying `tx2` Period**:
+    - `tx1` period is fixed at 20ms.
+    - `tx2` period is varied from 20ms to 100ms.
+    - We measure the impact on `tx1`'s latency and message interval.
+    - Since the periods `tx1` and `rx` are fixed, the message timings of `tx1`
+      must ideally not be impacted with varying `tx2` periods.
+
+2.  **Fixed `tx2` Period, Varying `tx1` Period**:
+    - `tx2` period is fixed at 20ms.
+    - `tx1` period is varied from 20ms to 100ms.
+    - We measure the impact on `tx2`'s latency and message interval.
+    - Since the periods `tx2` and `rx` are fixed, the message timings of `tx2`
+      must ideally not be impacted with varying `tx1` periods.
+
+## Results
+
+The results show that with the default buffer, the performance of one operator
+is heavily dependent on the other. However, with the async lock-free buffer,
+they are decoupled enabling true asynchronous execution of the operators.
+
+### TX1 Message Latency vs. TX2 Period
+
+![TX1 Latency vs. TX2 Period](tx1_latency_vs_tx2_period.png)
+
+With the default buffer, as `tx2`'s period increases, `tx1`'s maximum latency
+also increases linearly. Since `rx` cannot run before both the upstream
+operators (`tx1` and `tx2`) have written messages for it, the latency of `tx1`
+is affected by `tx2`. With the async lock-free buffer, `tx1`'s latency 
+remains consistent and low, regardless of `tx2`'s period. Therefore, async
+lock-free buffer unlocks independent connection between `tx1` and `rx`
+irrespective of the behavior of `tx2`.
+
+### IN1 Message Interval vs. TX2 Period
+
+![IN1 Message Interval vs. TX2 Period](in1_period_vs_tx2_period.png)
+
+Similarly, the maximum message interval for `tx1` (at the `in1` port of `rx`)
+increases with `tx2`'s period when using the default buffer.
+The async lock-free buffer keeps the interval stable.
+
+### TX2 Message Latency vs. TX1 Period
+
+![TX2 Latency vs. TX1 Period](tx2_latency_vs_tx1_period.png)
+
+The same trend is visible here. `tx2`'s message latency is affected by `tx1`'s period with the default buffer, but not with the async lock-free buffer.
+
+### IN2 Message Interval vs. TX1 Period
+
+![IN2 Message Interval vs. TX1 Period](in2_period_vs_tx1_period.png)
+
+The message interval for `tx2` (at the `in2` port of `rx`) remains stable with the async lock-free buffer, independent of `tx1`'s period.
+
+### Conclusion and Guidance
+
+The asynchronous lock-free buffer connection between the operators enables
+connected operators to run independently. When they are configured with 
+`SCHED_DEADLINE` Linux scheduling policy, they can run independently with 
+their respective runtime and periods. When the default buffer is used,
+operators may depend on other operator's runtime and periods, even when
+`SCHED_DEADLINE` policy is associated with them. Therefore, developers must
+take precaution when using `SCHED_DEADLINE` policy with the default buffer.
```

Comment:
Not sure I 100% follow the impact here, could you please clarify the behavior of operators when using the default buffer + `SCHED_DEADLINE`? Is the impact that operators will continue normal operation without performance benefit from `SCHED_DEADLINE`, or does this mean that there is risk of bad behavior where operators run independently when they are expected to rely on other operators' runtime+periods for correct performance of the default buffer?

Reply:
@tbirdso your first point is correct. When default buffers are used with `SCHED_DEADLINE`, operators will continue normal operation. However, they may not see any benefit of using `SCHED_DEADLINE` since they are dependent on other operator's behavior. It is also true that they may see degraded performance since operators have to satisfy multiple constraints when both default buffer and `SCHED_DEADLINE` are used. Let me clarify this in the README.